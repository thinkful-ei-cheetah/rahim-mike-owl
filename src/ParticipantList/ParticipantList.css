.participants-list {
  padding:10px;
  position:relative;
  max-height:calc(100vh - 45px - 75px);
  height:calc(100vh - 45px - 75px);
  overflow-y:scroll
}