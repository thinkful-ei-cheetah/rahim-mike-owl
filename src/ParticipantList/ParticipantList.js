import React from 'react';
import './ParticipantList.css';
import Participant from '../Participant/Participant';

function ParticipantList(props) {
  const list = props.store.participants;
  
  const inSessionParticipants = list.filter(participant => participant.inSession)
  const others = list.filter(participant => !participant.inSession)
  const orderedList = inSessionParticipants.concat(others)
  
  const participants = orderedList.map(participant => {
    return <Participant 
              avatar={participant.avatar}
              name={participant.name}
              inSession={participant.inSession}
              onStage={participant.onStage}
              key={participant.id} 
           />
  });

  return (
    <div className='participants-list'>
      {participants}
    </div>
  );
  
}


export default ParticipantList;
