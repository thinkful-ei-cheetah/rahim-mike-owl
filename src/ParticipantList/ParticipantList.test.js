import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ParticipantList from './ParticipantList';
import store from '../store';

describe('Participant List Component', () => {
  it('renders successfully', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <ParticipantList store={store} />,
      div
    )
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders a list of participants', () => {
    const tree = renderer.create(
      <ParticipantList store={store} />
    ).toJSON()
    expect(tree).toMatchSnapshot();
  });
})