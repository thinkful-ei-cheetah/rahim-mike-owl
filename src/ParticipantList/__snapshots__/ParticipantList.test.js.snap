// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participant List Component renders a list of participants 1`] = `
<div
  className="participants-list"
>
  <div
    className="session-participant"
  >
    <img
      alt="avatar"
      className="user-avatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
    <div
      className="participant-info"
    >
      <h6
        className="participant-name"
      >
        Koren Templeton
      </h6>
      <div
        className="participant-status"
      />
      <a
        href="#"
      >
        boot
      </a>
    </div>
  </div>
  <div
    className="session-participant"
  >
    <img
      alt="avatar"
      className="user-avatar"
      src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
    />
    <div
      className="participant-info"
    >
      <h6
        className="participant-name"
      >
        Caty Flucker
      </h6>
      <div
        className="participant-status"
      />
      <a
        href="#"
      >
        boot
      </a>
    </div>
  </div>
  <div
    className="session-participant"
  >
    <img
      alt="avatar"
      className="user-avatar"
      src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
    />
    <div
      className="participant-info"
    >
      <h6
        className="participant-name"
      >
        Frank Runciman
      </h6>
      <div
        className="participant-status"
      />
      <a
        href="#"
      >
        boot
      </a>
    </div>
  </div>
  <div
    className="session-participant"
  >
    <img
      alt="avatar"
      className="user-avatar"
      src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
    />
    <div
      className="participant-info"
    >
      <h6
        className="participant-name"
      >
        Ashla Attwool
      </h6>
      <div
        className="participant-status"
      />
      <a
        href="#"
      >
        boot
      </a>
    </div>
  </div>
  <div
    className="session-participant"
  >
    <img
      alt="avatar"
      className="user-avatar"
      src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
    />
    <div
      className="participant-info"
    >
      <h6
        className="participant-name"
      >
        Axe Kubicka
      </h6>
      <div
        className="participant-status"
      />
      <a
        href="#"
      >
        boot
      </a>
    </div>
  </div>
</div>
`;
